src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate.cs

```diff
@@ -72,6 +69,14 @@ public X509Certificate(string fileName, string password, X509KeyStorageFlags key
             Pal = CertificatePal.FromFile(fileName, password, keyStorageFlags);
         }
 
+        public X509Certificate(X509Certificate cert)
+        {
+            if (cert == null)
+                throw new ArgumentNullException(nameof(cert));
+
+            Pal = CertificatePal.FromHandle(cert.Handle); // Let FromHandle throw if certHandle is 0
```

Comment:
This doesn't work for Linux (it loses private keys).  You'll need to do a PAL clone.